packages/mobile/src/networkInfo/saga.ts

```diff
@@ -17,7 +17,7 @@ export function* waitForRehydrate() {
 
 export function* waitWeb3LastBlock() {
   yield waitForGethConnectivity()
-  yield checkWeb3Sync()
+  yield waitForWeb3Sync()
```

Comment:
@cmcewen IIRC you were advocated for calling this `check` but `wait` is what it's really doing. Mind if we rename?

Reply:
Volpe and I had a discussion about this - the reason I don't like using wait is that it actually makes a call to get the latest block to the geth node. Since I think we treat the `waitFor` generators as free in terms of cost, I would be wary of someone just adding this elsewhere and we end up doing a bunch of duplicate polling. I think `waitFor` should imply solely waiting on state changes.